llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1046,7 +1046,10 @@ class APFloat : public APFloatBase {
   ///
   /// \param Negative True iff the number should be negative.
   static APFloat getOne(const fltSemantics &Sem, bool Negative = false) {
-    return APFloat(Sem, Negative ? -1 : 1);
+    APFloat Val(Sem, 1U);
+    if (Negative)
+      Val.changeSign();
+    return Val;
```

Comment:
We probably should generalize this in terms of a generate integer power of 2 helper